tests/storage/conftest.py

```diff
@@ -1,7 +1,66 @@
+from __future__ import annotations
+
+import pytest
+
+import logging
+
+# explicit import for package-scope fixtures
+from pkgfixtures import pool_with_saved_yum_state
+
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from lib.host import Host
+
+QCOW2_REPO = "qcow2"
+QCOW2_REPO_FILE = f"/etc/yum.repos.d/xcp-ng-{QCOW2_REPO}.repo"
+
+
 def pytest_collection_modifyitems(config, items):
     # modify ordering so that ext is always tested first,
     # before more complex storage drivers
     for item in reversed(list(items)):
         if "_ext_" in item.path.name:
             items.remove(item)
             items.insert(0, item)
+
+
+@pytest.fixture(scope='package')
+def pool_with_qcow2(hostA1: Host, image_format):
+    import concurrent.futures
+    pool = hostA1.pool
+
+    # If we're testing VHD image format, skip installing qcow2 and just return the pool.
+    # Use yield+return because this is a generator fixture.
+    if image_format == "vhd":
+        yield pool
+        return
+
+    def check_qcow2_installed(host):
+        if host.file_exists(QCOW2_REPO_FILE):
+            raise Exception(
+                f'{QCOW2_REPO_FILE} is already installed on host {host}. This should not be the case.'
```

Comment:
I see where this comes from, but that doesn't work here. We want to keep the ability to install any packages on the test hosts, and the test must not tell us what to do here. It's different from the xostor case which is already a production feature and is tested using what's in the dedicated test repositories.